File: app-ios/ios/service/AccountService.m
//
//  AccountService.m
//  ios
//
//  Created by Peter Zhan on 2017/10/16.
//  Copyright © 2017年 com.enuocms. All rights reserved.
//

#import "AccountService.h"

@implementation AccountService
-(void)login:(NSString*)username
    password:(NSString*)password
     success:(APIObjectHandle)success_callback
     failure:(APIErrorHandle)failure_callback{

    NSMutableDictionary *parameters = [self UserInfoparameters];
    [parameters setObject:username  forKey:@"username"];
    [parameters setObject:password  forKey:@"password"];
    
    [self PostRequestWithFunction:@"login" parameters:parameters success:^(APIResult *result, NSDictionary *dict) {
        
        UserInfo *info = nil;
        if(result.success){
            NSDictionary *data = [dict objectForKey:@"data"];
            info = [data toModel:[UserInfo class]];
            info.ID = [data stringForKey:@"id"];
        }
        
        if (success_callback) {
            success_callback(result,info);
        }
    } failure:failure_callback];
}
@end

File: app-swift/ios/ios/service/AccountService.swift
//
//  AccountService.swift
//  ios
//
//  Created by Peter Zhan on 2017/10/20.
//  Copyright © 2017年 com.enuocms. All rights reserved.
//

import Foundation

class AccountService: BaseService {
    func login(username:String,password:String,success:@escaping APIObjectHandle,failure:@escaping APIErrorHandle) {
        var parameters = self.UserInfoparameters()
        parameters["username"] = username
        parameters["password"] = password
        
        self.RequestWithFunction(url: "login", parameters: parameters, method: .post, successHandle: { (result, json) in
            var info:UserInfo? = nil
            if(result.success){
                info = UserInfo.deserialize(from: json["data"].rawString())
            }
            success(result,info)
        }, failureHandle: failure)
    }
}
